src/gdbserver/gdb-server.c

```diff
@@ -38,7 +38,7 @@
 static stlink_t *connected_stlink = NULL;
 static bool semihosting = false;
 static bool serial_specified = false;
-static char serialnumber[28] = {0};
+static uint8_t serialnumber[STLINK_SERIAL_SIZE] = {0}; // XXX? Why was size 28?
```

Comment:
Got it, thanks.
Anyway, in code is've seen different serial sizes - from 16 to 28... but seems all of them is incorrect :)

I've found the following document http://www.st.com/resource/en/application_note/cd00169020.pdf
where described that serial could have custom length (didn't said maximum exactly) and ascii representation can be greater than 28, e.g. on my STM32F407G-DISC1, the key have 48 bytes in ascii representation and struct stlink_t with buffer serial[16] (as it was) can't contain all data of libusb_get_string_descriptor_ascii call in stlink_open_usb. I'm working on it.